[ML] after upgrade to 7.11 BC1, real-time AD job state turns to "closed"

**Steps to reproduce:**

0) On cloud production environment (EC_REGION=gcp-europe-west1),
1) Set up a 7.10.0 or 7.9.0 cluster with ml node enabled
2) create several real-time AD jobs. then leave them in opened and started state
3) upgrade to 7.11.0 BC1
4) after upgrade finished, check those jobs being created in step 2,

**Observations:**
Jobs are in **closed** and started state.

**Expected:**
Jobs should be still in **opened** and started state. 

Message from one job:
![image](https://user-images.githubusercontent.com/47184485/102905640-774c8a80-4441-11eb-83dd-032521b38b5d.png)


WARN exception in es.log:
```
[instance-0000000005] task job-airplane_1605130010_000_0 failed with an exception
org.elasticsearch.transport.RemoteTransportException: [instance-0000000000][172.17.0.19:19174][cluster:admin/xpack/ml/job/model_snapshots/revert]
Caused by: org.elasticsearch.ElasticsearchStatusException: Can only revert to a model snapshot when the job is closed.
	at org.elasticsearch.xpack.core.ml.utils.ExceptionsHelper.conflictStatusException(ExceptionsHelper.java:67) ~[?:?]
	at org.elasticsearch.xpack.ml.action.TransportRevertModelSnapshotAction.lambda$masterOperation$1(TransportRevertModelSnapshotAction.java:100) ~[?:?]
	at org.elasticsearch.action.ActionListener$1.onResponse(ActionListener.java:63) ~[elasticsearch-7.11.0.jar:7.11.0]
	at org.elasticsearch.xpack.ml.job.persistence.JobConfigProvider$5.onResponse(JobConfigProvider.java:399) ~[?:?]

```

--- a/x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/task/OpenJobPersistentTasksExecutor.java
+++ b/x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/task/OpenJobPersistentTasksExecutor.java
@@ -12,6 +12,7 @@ import org.apache.logging.log4j.message.ParameterizedMessage;
 import org.elasticsearch.ElasticsearchException;
 import org.elasticsearch.ElasticsearchStatusException;
 import org.elasticsearch.ResourceNotFoundException;
+import org.elasticsearch.Version;
 import org.elasticsearch.action.ActionListener;
 import org.elasticsearch.client.Client;
 import org.elasticsearch.cluster.ClusterState;
@@ -61,6 +62,10 @@ import static org.elasticsearch.xpack.core.ml.MlTasks.AWAITING_UPGRADE;
 public class OpenJobPersistentTasksExecutor extends AbstractJobPersistentTasksExecutor<OpenJobAction.JobParams> {
 
     private static final Logger logger = LogManager.getLogger(OpenJobPersistentTasksExecutor.class);
+
+    // Resuming a job with a running datafeed from its current snapshot was added in 7.11 and
+    // can only be done if the master node is on or after that version.
+    private static final Version MIN_MASTER_NODE_VERSION_FOR_REVERTING_TO_CURRENT_SNAPSHOT = Version.V_7_11_0;
 
     public static String[] indicesOfInterest(String resultsIndex) {
         if (resultsIndex == null) {
@@ -205,7 +210,9 @@ public class OpenJobPersistentTasksExecutor extends AbstractJobPersistentTasksEx
 
         ActionListener<Boolean> hasRunningDatafeedTaskListener = ActionListener.wrap(
             hasRunningDatafeed -> {
-                if (hasRunningDatafeed) {
+                if (hasRunningDatafeed && clusterState.nodes().getMasterNode().getVersion().onOrAfter(
+                    MIN_MASTER_NODE_VERSION_FOR_REVERTING_TO_CURRENT_SNAPSHOT)) {
+
                     // This job has a running datafeed attached to it.
                     // In order to prevent gaps in the model we revert to the current snapshot deleting intervening results.
                     revertToCurrentSnapshot(jobTask.getJobId(), ActionListener.wrap(response -> openJob(jobTask), jobTask::markAsFailed));

